src/assets.zig

```diff
@@ -271,21 +273,35 @@ fn registerItem(assetFolder: []const u8, id: []const u8, zon: ZonElement) !*item
 		texturePath = try std.fmt.bufPrint(&buf1, "{s}/{s}/items/textures/{s}", .{assetFolder, mod, texture});
 		replacementTexturePath = try std.fmt.bufPrint(&buf2, "assets/{s}/items/textures/{s}", .{mod, texture});
 	}
-	return items_zig.register(assetFolder, texturePath, replacementTexturePath, id, zon);
+	_ = items_zig.register(assetFolder, texturePath, replacementTexturePath, id, zon);
 }
 
 fn registerTool(assetFolder: []const u8, id: []const u8, zon: ZonElement) void {
 	items_zig.registerTool(assetFolder, id, zon);
 }
 
 fn registerBlock(assetFolder: []const u8, id: []const u8, zon: ZonElement) !void {
-	const block = blocks_zig.register(assetFolder, id, zon);
+	if(zon == .null) std.log.err("Missing block: {s}. Replacing it with default block.", .{id});
+
+	_ = blocks_zig.register(assetFolder, id, zon);
 	blocks_zig.meshes.register(assetFolder, id, zon);
+}
 
-	if(zon.get(bool, "hasItem", true)) {
-		const item = try registerItem(assetFolder, id, zon.getChild("item"));
-		item.block = block;
+fn assignBlockItem(assetFolder: []const u8, stringId: []const u8, zon: ZonElement, itemPalette: *Palette) !void {
+	if(!zon.get(bool, "hasItem", true)) return;
+
+	if(!items_zig.hasRegistered(stringId)) {
```

Comment:
This check should only be made when trying to register new items, not when registering items from the palette to avoid problems with duplicate entries (produced by migrating multiple items onto one id).